src/storage/lock_manager/util.rs

```diff
@@ -2,32 +2,45 @@
 
 use super::Lock;
 use crate::storage::mvcc::Error as MvccError;
-use crate::storage::txn::Error as TxnError;
+use crate::storage::txn::{Error as TxnError, ProcessResult};
 use crate::storage::Error as StorageError;
 use crate::storage::Key;
-use std::collections::hash_map::DefaultHasher;
-use std::hash::{Hash, Hasher};
+use farmhash;
 
 pub fn extract_lock_from_result(res: &Result<(), StorageError>) -> Lock {
     match res {
         Err(StorageError::Txn(TxnError::Mvcc(MvccError::KeyIsLocked { key, ts, .. }))) => Lock {
             ts: *ts,
             hash: gen_key_hash(&Key::from_raw(&key)),
         },
-        _ => panic!("unsupported mvcc error"),
+        _ => panic!("unexpected mvcc error"),
     }
 }
 
+// TiDB uses the same hash algorithm.
 pub fn gen_key_hash(key: &Key) -> u64 {
-    let mut s = DefaultHasher::new();
-    key.hash(&mut s);
-    s.finish()
+    farmhash::fingerprint64(&key.to_raw().unwrap())
 }
 
 pub fn gen_key_hashes(keys: &[Key]) -> Vec<u64> {
     keys.iter().map(|key| gen_key_hash(key)).collect()
 }
 
+pub fn extract_raw_key_from_process_result(pr: &ProcessResult) -> &[u8] {
+    match pr {
+        ProcessResult::MultiRes { results } => {
```

Comment:
I think we don't need to panic if failed to extract the key, as it will only be used for logging.

Reply:
It never occurs. If panicked, it's a bug in implementation.